tensorflow/src/TensorFlow/Types.hs

```diff
@@ -95,66 +100,31 @@ import Proto.Tensorflow.Core.Framework.Types (DataType(..))
 import TensorFlow.Internal.VarInt (getVarInt, putVarInt)
 import qualified TensorFlow.Internal.FFI as FFI
 
--- | Data about a tensor that is encoded for the TensorFlow APIs.
-newtype TensorData a = TensorData { unTensorData :: FFI.TensorData }
-
 -- | The class of scalar types supported by tensorflow.
-class TensorType a where
+class TensorDataType V.Vector a => TensorType a where
```

Comment:
This particular superclass constraint seems a little odd since I think for most types users will want to use S.Vector rather than V.Vector...does anything get more complicated if you just remove it?  (Since otherwise TensorType is usually about graph construction, not fetching.)